File: 0495.提莫攻击/0495-提莫攻击.go
func findPoisonedDuration(timeSeries []int, duration int) int {
    ret := 0
    i := 0
    for i < len(timeSeries) - 1 {
        ret += min(duration, timeSeries[i+1]-timeSeries[i])
        i += 1
    }
    if len(timeSeries) > 0 {
        ret += duration
    }
    
    return ret
 }

func min(a, b int) int {
    if a < b {
        return a
    } else {
        return b
    }
}

File: 0495.提莫攻击/0495-提莫攻击.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int ret = 0;
        int i = 0;
        while (i < timeSeries.length - 1) {
            ret += Math.min(duration, timeSeries[i+1] - timeSeries[i]);
            i += 1;
        }
        if (timeSeries.length > 0) {
            ret += duration;
        }
        
        return ret;
    }
}